### PR TITLE
Add try with resource support

### PR DESCRIPTION
**Pull request by JDLogic**
Attempt at resolving #17. This should not introduce any errors as far I can tell.
[Resulting MC Diff](https://gist.github.com/JDLogic/0ef03494be8fb30d96242c08271356bc)

### CODE REVIEW

File: FernFlower-Patches/0024-Add-try-with-resource-support.patch [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
++import org.jetbrains.java.decompiler.modules.decompiler.stats.CatchStatement;
++import org.jetbrains.java.decompiler.modules.decompiler.stats.RootStatement;
++import org.jetbrains.java.decompiler.modules.decompiler.stats.Statement;
++import org.jetbrains.java.decompiler.struct.gen.VarType;
++import java.util.ArrayList;
++import java.util.List;
++
++public class TryHelper
++{
++  private static final String[] CLOSABLES = {
```
Comment (kashike) [inline FernFlower-Patches/0024-Add-try-with-resource-support.patch]: `CLOSEABLES`

Comment (bs2609) [reply FernFlower-Patches/0024-Add-try-with-resource-support.patch]: Does this need a list or can it work like #4 and check for `AutoCloseable` instances?

Comment (kashike) [reply FernFlower-Patches/0024-Add-try-with-resource-support.patch]: ...yes, it should do what #4 does now that I think about it.

Comment (JDLogic) [reply FernFlower-Patches/0024-Add-try-with-resource-support.patch]: Yeah, that was a derp on my part. Only the one instance check is needed. I’ll clean that up when I get a chance.

**Comment by JDLogic** [comment]
Updated the diff in the first comment to reflect the new changes.

**Comment by JDLogic** [comment]
I made sure this has no conflicts, and I’m happy with the changes. So this is ready for re-review.

**Pull request merged**